src/Aspire.Hosting/Kafka/KafkaBuilderExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using System.Text.Json;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+public static class KafkaBuilderExtensions
+{
+    /// <summary>
+    /// Adds a Kafka broker container to the application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port of Kafka broker.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{KafkaContainerResource}"/></returns>
+    public static IResourceBuilder<KafkaContainerResource> AddKafkaBrokerContainer(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    {
+        const int kafkaBrokerPort = 9092;
+        port ??= kafkaBrokerPort;
+
+        return builder.AddResource(new KafkaContainerResource(name))
+             .WithAnnotation(new ManifestPublishingCallbackAnnotation((json) => json.WriteString("type", "kafka.server.v0")))
+             .WithAnnotation(new ServiceBindingAnnotation(ProtocolType.Tcp, port: port, containerPort: kafkaBrokerPort))
+             .WithAnnotation(new ContainerImageAnnotation
+             {
+                 Image = "confluentinc/confluent-local",
+                 Tag = "latest"
+             })
+             .WithEnvironment("KAFKA_LISTENERS", "PLAINTEXT://localhost:29092,CONTROLLER://localhost:29093,PLAINTEXT_HOST://0.0.0.0:9092")
+             .WithEnvironment("KAFKA_ADVERTISED_LISTENERS", $"PLAINTEXT://localhost:29092,PLAINTEXT_HOST://localhost:{port}");
```

Comment:
Where do these ports come from? Is it worth leaving a comment about them?

Reply:
I added comments and links. These ports are the defaults of the `confluentinc/confluent-local` image which are set when the container starts if the env var is not supplied. See https://github.com/confluentinc/kafka-images/blob/master/local/include/etc/confluent/docker/configureDefaults